erpnext/accounts/doctype/dunning/dunning.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.model.document import Document
+import json
+from six import string_types
+from frappe.utils import getdate, get_datetime, rounded, flt
+from erpnext.loan_management.doctype.loan_interest_accrual.loan_interest_accrual import days_in_year
+
+
+class Dunning(Document):
+    def validate(self):
+        self.validate_overdue_days()
+        self.validate_amount()
+
+    def validate_overdue_days(self):
+        self.overdue_days = (getdate(self.posting_date) -
+                             getdate(self.due_date)).days or 0
+
+    def validate_amount(self):
+        amounts = calculate_interest_and_amount(
+            self.posting_date, self.outstanding_amount, self.rate_of_interest, self.dunning_fee, self.overdue_days)
+        if self.interest_amount != amounts.get('interest_amount'):
+            self.interest_amount = amounts.get('interest_amount')
+        if self.dunning_amount != amounts.get('dunning_amount'):
+            self.dunning_amount = amounts.get('dunning_amount')
+        if self.grand_total != amounts.get('grand_total'):
+            self.grand_total = amounts.get('grand_total')
+
+
+def resolve_dunning(doc, state):
+    for reference in doc.references:
+        if reference.reference_doctype == 'Sales Invoice' and reference.outstanding_amount <= 0:
+            dunnings = frappe.get_list('Dunning', filters={
+                                       'sales_invoice': reference.reference_name, 'status': ('!=', 'Resolved')})
+
+            for dunning in dunnings:
+                dunning_doc = frappe.get_doc('Dunning', dunning.name)
+                dunning_doc.status = 'Resolved'
+                dunning_doc.save()
+
+
+@frappe.whitelist()
+def get_dunning_letter_text(dunning_type, doc, language=None):
+    if isinstance(doc, string_types):
+        doc = json.loads(doc)
+        if language:
+            filters = {'parent': dunning_type,
+                       'language': language}
+        else:
+            filters = {'parent': dunning_type,
+                       'is_default_language': 1}
+    letter_text = frappe.db.get_value('Dunning Letter Text',
+                                      filters,
+                                      ['body_text', 'closing_text', 'language'], as_dict=1)
```

Comment:
```suggestion
    letter_text = frappe.db.get_value('Dunning Letter Text',
        filters, ['body_text', 'closing_text', 'language'], as_dict=1)
```

We generally used this kind of indentation everywhere. Can you please fix this in all places?